WARNINGS due to register class

We have WARNINGS on non-linux in http://bioconductor.org/checkResults/devel/bioc-LATEST/mzR/veracruz2-checksrc.html
due to `boost/libs/regex/src/regex_raw_buffer.cpp:39:4: warning: 'register' storage class specifier is deprecated and incompatible with C++1z [-Wdeprecated-register]`
which can be fixed the same way we did it back in 
https://github.com/sneumann/mzR/commit/c8d2d5db0c0a1821e1465bba68e69131a66eb218

--- a/src/boost/libs/regex/src/regex_raw_buffer.cpp
+++ b/src/boost/libs/regex/src/regex_raw_buffer.cpp
@@ -36,15 +36,15 @@ namespace boost{ namespace re_detail{
 
 void BOOST_REGEX_CALL raw_storage::resize(size_type n)
 {
-   register size_type newsize = start ? last - start : 1024;
+   size_type newsize = start ? last - start : 1024;
    while(newsize < n)
       newsize *= 2;
-   register size_type datasize = end - start;
+   size_type datasize = end - start;
    // extend newsize to WORD/DWORD boundary:
    newsize = (newsize + padding_mask) & ~(padding_mask);
 
    // allocate and copy data:
-   register pointer ptr = static_cast<pointer>(::operator new(newsize));
+   pointer ptr = static_cast<pointer>(::operator new(newsize));
    BOOST_REGEX_NOEH_ASSERT(ptr)
    if(start)
       std::memcpy(ptr, start, datasize);
@@ -63,7 +63,7 @@ void* BOOST_REGEX_CALL raw_storage::insert(size_type pos, size_type n)
    BOOST_ASSERT(pos <= size_type(end - start));
    if(size_type(last - end) < n)
       resize(n + (end - start));
-   register void* result = start + pos;
+   void* result = start + pos;
    std::memmove(start + pos + n, start + pos, (end - start) - pos);
    end += n;
    return result;

